### PR TITLE
hide fusion gene table in the summary view if no fusion data present

### PR DESCRIPTION
**Pull request by khzhu**
Resolve cBioPortal/cbioportal [https://github.com/cBioPortal/cbioportal/issues/6709](url)

Describe changes proposed in this pull request:
- [x] fetch numberOfFusionProfiledSamples/numberOfFusionUnprofiledSamples from API
- [x] hide fusion gene table if no fusion gene present
- [x] use numberOfFusionProfiledSamples for sample profiled in the fusion gene table

# Checks
- [ ] Has tests or has a separate issue that describes the types of test that should be created. If no test is included it should explicitly be mentioned in the PR why there is no test.
- [ ] The commit log is comprehensible. It follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [ ] Is this PR adding logic based on one or more **clinical** attributes? If yes, please make sure validation for this attribute is also present in the data validation / data loading layers (in backend repo) and documented in [File-Formats Clinical data section](https://github.com/cBioPortal/cbioportal/blob/master/docs/File-Formats.md#clinical-data)!

# Any screenshots or GIFs?
If this is a new visual feature please add a before/after screenshot or gif
here with e.g. [GifGrabber](http://www.gifgrabber.com/).

# Notify reviewers
Read our [Pull request merging
policy](../CONTRIBUTING.md#pull-request-merging-policy). It can help to figure out who worked on the
file before you. Please use `git blame <filename>` to determine that
and notify them either through slack or by assigning them as a reviewer on the PR

### CODE REVIEW

File: src/pages/studyView/StudyViewPageStore.ts [hunk 1]
```diff
@@ -2486,16 +2486,18 @@ export class StudyViewPageStore {
...
-                uniqueKey: UniqueKey.FUSION_GENES_TABLE,
-                dataType: getChartMetaDataType(UniqueKey.FUSION_GENES_TABLE),
-                patientAttribute:false,
-                displayName: 'Fusion Genes',
-                priority: getDefaultPriorityByUniqueKey(UniqueKey.FUSION_GENES_TABLE),
-                renderWhenDataChange: true,
-                description: ''
-            };
+        if (!_.isEmpty(this.molecularProfileSampleCounts.result)) {
+            if (this.molecularProfileSampleCounts.result!.numberOfFusionProfiledSamples > 0) {
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts]: not important but might as well just add this to wrapping if statement

Comment (alisman) [reply src/pages/studyView/StudyViewPageStore.ts]: i.e. if (A && B) {  }

note that B will not evaluate if A is false

File: src/pages/studyView/StudyViewPageStore.ts [hunk 2]
```diff
@@ -4093,6 +4098,19 @@ export class StudyViewPageStore {
         default: []
     })
 
+    private withFusionDataChartCounts = remoteData<ClinicalDataCountSummary[]>({
+        invoke: async () => {
+            let molecularProfileSampleCounts = await internalClient.fetchMolecularProfileSampleCountsUsingPOST({
+                studyViewFilter: Object.assign({}, this.filters, { withFusionData: undefined })
+            });
+            return Promise.resolve(
```
Comment (alisman) [inline src/pages/studyView/StudyViewPageStore.ts]: i'm pretty sure you don't need to Promise.resolve.  Just return getClinicalDataCountWithColorByCategoryCounts(...)

File: src/pages/studyView/StudyViewPageStore.ts (hunk 1 above)
Comment (khzhu) [reply src/pages/studyView/StudyViewPageStore.ts]: good point. change made.

File: src/pages/studyView/StudyViewPageStore.ts (hunk 2 above)
Comment (khzhu) [reply src/pages/studyView/StudyViewPageStore.ts]: got it. changed. thanks!

**Comment by alisman** [comment]
@khzhu looks like there are a lot of ts errors causing build to fail

**Comment by khzhu** [comment]
Hi @alisman , this PR needs backend changes. /sample-counts/fetch web service endpoint needs to be updated.

**Comment by khzhu** [comment]
[backend PR](https://github.com/cBioPortal/cbioportal/pull/6731), @alisman .

**Comment by khzhu** [comment]
Hi @sheridancbio , would you be able to review this PR. It will fix the [problem](https://github.com/cBioPortal/cbioportal/issues/6709) reported by Tali. Thank you!

File: src/pages/studyView/StudyViewPageStore.ts [hunk 3]
```diff
@@ -2486,15 +2486,16 @@ export class StudyViewPageStore {
...
-        if (!_.isEmpty(this.mutationProfiles.result)) {
-            _chartMetaSet[UniqueKey.FUSION_GENES_TABLE] = {
-                uniqueKey: UniqueKey.FUSION_GENES_TABLE,
-                dataType: getChartMetaDataType(UniqueKey.FUSION_GENES_TABLE),
-                patientAttribute:false,
-                displayName: 'Fusion Genes',
-                priority: getDefaultPriorityByUniqueKey(UniqueKey.FUSION_GENES_TABLE),
-                renderWhenDataChange: true,
-                description: ''
+        if ((!_.isEmpty(this.molecularProfileSampleCounts.result)) &&
```
Comment (zhx828) [inline src/pages/studyView/StudyViewPageStore.ts]: I don't think this condition is needed anymore

Comment (khzhu) [reply src/pages/studyView/StudyViewPageStore.ts]: webstorm will throw you an error if you remove the first nullity check.

**Comment by khzhu** [comment]
Hi @zhx828 , may I please have your final review? thank you!

**Comment by alisman** [comment]
@khzhu i now understand api changes haven't been merged yet.

there are some ts errors.  could you fix them and rebase this.  thanks

**Comment by khzhu** [comment]
hi Aaron, did rebase and fixed errors. here is [backend PR](https://github.com/cBioPortal/cbioportal/pull/6731), needs an approval from Benjamin or Rob. Thank you!

**Comment by alisman** [comment]
@alisman we need to backfill data (populate fusion geneticprofile) before we can merge this because we would otherwise hide data

**Comment by inodb** [comment]
This has been merged to release-3.2.0 branch since it required db migrations

**Pull request closed without merging**